### PR TITLE
Implemented New Getters

### PR DESCRIPTION
**Pull request by Transyltooniaa**
Before submitting a pull request (PR), please read the [contributing guide](https://github.com/neuroinformatics-unit/.github/blob/main/CONTRIBUTING.md).

Please fill out as much of this template as you can, but if you have any problems or questions, just leave a comment and we will help out :)

## Description

**What is this PR**

- [ ] Bug fix
- [ ] Addition of a new feature
- [*] Other

**Why is this PR needed?**
 Adds convenience functions like getter / setter for paths

**What does this PR do?**
This PR introduces two new methods to streamline folder lookups within the datashuttle project:

get_all_sub_and_ses

Returns a list of dictionaries containing each subject and the corresponding session paths found in the project.
Useful for enumerating all subject-session pairs, including optional support for local-only vs. local+central searches.
find_folders

Finds all folders matching a given glob pattern under specified subject(s) and session(s).
Also supports the “all_ses” keyword to search across all sessions for a given subject.

## References

Please reference any existing issues/PRs that relate to this PR.

## How has this PR been tested?
Integration Tests: Created tests to verify that both methods correctly enumerate directories in local-only and full (local+central) project modes.
Edge Cases: Confirmed behavior when no sessions exist, when multiple sessions exist, and when patterns match no folders.

Please explain how any new code has been tested, and how you have ensured that no existing functionality has changed.

## Is this a breaking change?
No

If this PR breaks any existing functionality, please explain how and why.

## Does this PR require an update to the documentation?
No

If any features have changed, or have been added. Please explain how the
documentation has been updated.

## Checklist:

- [*] The code has been tested locally
- [ ] Tests have been added to cover all new functionality
- [ ] The documentation has been updated to reflect any changes
- [ ] The code has been formatted with [pre-commit](https://pre-commit.com/)

### CODE REVIEW

**Comment by Transyltooniaa** [comment]
Hey , This is my first pull request, to get started with the project, I have tried to implement this issue #442 .  Please let me know if this is correct and let me know if anything else was expected out of this issue so that I can build upon this.

Apologies for any inconvenience caused

File: datashuttle/utils/getters.py [hunk 1]
```diff
@@ -339,3 +340,86 @@ def get_all_sub_and_ses_names(
...
+    """
+    project_info = []
+    sub_ses_names = get_all_sub_and_ses_names(
+        self.cfg, cast(TopLevelFolder, top_level_folder), local_only
+    )
+
+    for sub in sub_ses_names["sub"]:
+        session_paths = []
+
+        if local_only:
```
Comment (JoeZiminski) [inline datashuttle/utils/getters.py:368]: This should check the dict within `sub_ses_names["sub"]` which I believe contains information on central vs local. For central, this would be `get_central_path()`.

File: datashuttle/utils/getters.py [hunk 2]
```diff
@@ -339,3 +340,86 @@ def get_all_sub_and_ses_names(
             )
 
     return {"sub": all_sub_folder_names, "ses": all_ses_folder_names}
+
+
+def get_all_sub_and_ses(
```
Comment (JoeZiminski) [inline datashuttle/utils/getters.py:345]: Thanks for this @Transyltooniaa! In a recent PR (due to be merged soon, #464) this functionality was added to `get_all_sub_and_ses_names` which is now called `get_all_sub_and_ses_paths`. This will be merged soon and we can take another look at this. Please feel free to start a discussion on any issues to brainstorm an approach to avoid clashes, apologies for this I still hope coding this up was useful.

File: datashuttle/utils/getters.py [hunk 3]
```diff
@@ -339,3 +340,86 @@ def get_all_sub_and_ses_names(
...
+    top_folder_literal = cast(TopLevelFolder, top_level_folder)
+
+    if session_scope == "all_ses":
+        ses_names = get_all_sub_and_ses_names(
+            self.cfg, top_folder_literal, local_only
+        )["ses"].get(subject, [])
+    else:
+        ses_names = session_scope
+
+    search_paths = []
```
Comment (JoeZiminski) [inline datashuttle/utils/getters.py:410]: This is a good approach, I think this will be easier to implement when #464 is merged

**Review by JoeZiminski — COMMENTED** [review]
Hi @Transyltooniaa thanks for this this is a nice feature addition. There are some changes in #464 which will make this implementation simpler (please see comments). This will be merged shortly, I will get back to you ASAP.

File: datashuttle/utils/getters.py (hunk 3 above)
Comment (JoeZiminski) [reply datashuttle/utils/getters.py:410]: #464 is now merged, I think now you will be able to take the paths directly from this function.

**Comment by JoeZiminski** [comment]
Hi @Transyltooniaa do you think you will have time to look into this further? No worries if not! It will be a useful template for tackling this issue in future.

**Comment by Transyltooniaa** [comment]
Apologies for the delay. Yes, I intend to complete this PR. Once my university exams are finished this Monday, I can work on this
